cmd/kaspaminer/mineloop.go

```diff
@@ -45,15 +48,7 @@ func mineLoop(client *minerClient, numberOfBlocks uint64, targetBlocksPerSecond
 		blockInWindowIndex := 0
 
 		for i := uint64(0); numberOfBlocks == 0 || i < numberOfBlocks; i++ {
-
-			foundBlock := make(chan *externalapi.DomainBlock)
-			mineNextBlock(client, miningAddr, foundBlock, mineWhenNotSynced, templateStopChan, errChan)
-			block := <-foundBlock
-			templateStopChan <- struct{}{}
-			err := handleFoundBlock(client, block)
-			if err != nil {
-				errChan <- err
-			}
+			foundBlockChan <- mineNextBlock(mineWhenNotSynced)
```

Comment:
This implicitly blocks if 100 blocks were solved but submitBlock takes a long time. Is this intentional?

Reply:
Yes